### ISSUE TITLE
KD src compile errors in Qt Creator 6

### ISSUE THREAD

**Issue by mintykat** [KDAB/KDReports#34]
I create a widget project in Qt Creator and added the source code modules to the .pro file. I'm getting 4 errors, all the same issue:
C:\dev\VIDS-6001\Projects\cpp\apps\qKD\build-kdtest-Desktop_Qt_6_2_3_MSVC2019_64bit-Debug\debug\moc_KDReportsReport.cpp:89: error: C2491: 'KDReports::Report::staticMetaObject': definition of dllimport static data member not allowed - 
This is the code:
const QMetaObject KDReports::Report::staticMetaObject = { {
    QMetaObject::SuperData::link<QObject::staticMetaObject>(),
    qt_meta_stringdata_KDReports__Report.offsetsAndSize,
    qt_meta_data_KDReports__Report,
    qt_static_metacall,
    nullptr,
qt_incomplete_metaTypeArray<qt_meta_stringdata_KDReports__Report_t
, QtPrivate::TypeAndForceComplete<QString, std::true_type>, QtPrivate::TypeAndForceComplete<Report, std::true_type>

**Comment by dfaure-kdab** [comment]
You're experiencing problem because you're bypassing the buildsystem provided with KDReports, and the export macros in KDReports assume it will be built as a shared library.

Solution: try adding this to your .pro file:

    DEFINES += KDREPORTS_STATICLIB

If this is a new project, however, I recommend using CMake.
See https://www.youtube.com/watch?v=952DsSbAHHg&list=PL6CJYn40gN6g1_yY2YkqSym7FWUid926M for more details.

**Comment by mintykat** [comment]
Thanks, I added KDREPORTS_STATICLIB to the defines in qt creator and now it compiles. Now I'm getting a lot of link errors as:
KDReportsAutoTableElement.obj:-1: error: LNK2019: unresolved external symbol "__declspec(dllimport) protected: __cdecl KDReports::AbstractTableElement::AbstractTableElement(void)" (__imp_??0AbstractTableElement@KDReports@@IEAA@XZ) referenced in function "public: __cdecl KDReports::AutoTableElement::AutoTableElement(class QAbstractItemModel *)" (??0AutoTableElement@KDReports@@QEAA@PEAVQAbstractItemModel@@@Z)
Now, I'll look at any libs I have to add to qt creator.

**Comment by dfaure-kdab** [comment]
You must have missed some files, like KDReportsAbstractTableElement.cpp?

I just tested `qmake -project` in the src/KDReports directory and adding a QT+= line, it builds (except for main() missing, obviously). See [.pro file attached.](http://www.kdab.com/~dfaure/2022/KDReports.pro)

**Comment by mintykat** [comment]
Thx. Yes, I left that file out, so now I'm getting fewer errors, probably I did not list them all, ex:
KDReportsAutoTableElement.obj:-1: error: LNK2019: unresolved external symbol "__declspec(dllimport) public: class QTextDocument & __cdecl KDReports::ReportBuilder::currentDocument(void)" (__imp_?currentDocument@ReportBuilder@KDReports@@QEAAAEAVQTextDocument@@XZ) referenced in function "public: virtual void __cdecl KDReports::AutoTableElement::build(class KDReports::ReportBuilder &)const " (?build@AutoTableElement@KDReports@@UEBAXAEAVReportBuilder@2@@Z)

**Comment by dfaure-kdab** [comment]
I don't understand. Surely you don't need my help to copy the complete list of cpp files?

**Comment by mintykat** [comment]
No, but you pointed me in the right direction. I'm double checking.

**Comment by mintykat** [comment]
By the way, a simple clean operation straightened out all my link errors and I got a good build - just thought I'd point that out as it could be helpful to other victims. I was watching your interesting youtubes on cmake. Evidentally, my new Qt Creator version 6.0.2 uses cmake as you pointed out. Looking forward to making reports. Thx.
